Doc/library/asyncio-task.rst

```diff
@@ -1067,14 +1067,69 @@ Scheduling From Other Threads
    This function is meant to be called from a different OS thread
    than the one where the event loop is running.  Example::
 
-     # Create a coroutine
-     coro = asyncio.sleep(1, result=3)
-
-     # Submit the coroutine to a given loop
-     future = asyncio.run_coroutine_threadsafe(coro, loop)
-
-     # Wait for the result with an optional timeout argument
-     assert future.result(timeout) == 3
+     # Create a contextvar in your module
```

Comment:
Can we keep this sample minimal? It can as simple as creating a function which calls `asyncio.sleep` with `result = hello world` and call that function from a thread.

Reply:
I think it's important to show how to pass the loop to to_thread, and that you should be using run_coroutine_threadsafe to interact with blocking IO

Reply:
```py
import asyncio
from threading import Thread


def func(loop):
    fut = asyncio.run_coroutine_threadsafe(asyncio.sleep(1, result='hello'), loop)
    print(fut.result())

async def main():
    loop = asyncio.get_running_loop()
    t = Thread(target=lambda: func(loop))
    t.start()
    # perform other operations
    await asyncio.sleep(2)

asyncio.run(main())
```

I would like something like this.

Reply:
No it should definitely use asyncio.to_thread

Reply:
> No it should definitely use asyncio.to_thread

```py
import asyncio


def func(loop):
    fut = asyncio.run_coroutine_threadsafe(asyncio.sleep(1, result='hello'), loop)
    print(fut.result())

async def main():
    loop = asyncio.get_running_loop()
    await asyncio.to_thread(func, loop)

asyncio.run(main())
```
That was just an example, if you want to to_thread then how about this?

Reply:
Hmm maybe we can just get to_thread to just set the default loop for run_coroutine_threadsafe

Reply:
I don't think we should be changing it, these APIs are rare to use and explicit loop is better to pass here. let's just make the docs change here.